google-cloud-spanner/src/main/java/com/google/cloud/spanner/StructReader.java

```diff
@@ -132,6 +132,16 @@
   /** Returns the value of a non-{@code NULL} column with type {@link Type#date()}. */
   Date getDate(String columnName);
 
+  /** Returns the value of a non-{@code NULL} column as a {@link Value}. */
+  default Value getValue(int columnIndex) {
+    throw new UnsupportedOperationException("method should be overwritten");
+  }
+
+  /** Returns the value of a non-{@code NULL} column as a {@link Value}. */
```

Comment:
nit: This should be updated to indicate that this method can be called for columns that contain a null value, and what the behavior is when the value is null.